### ISSUE TITLE
fail_on_error works with lint warnings

### ISSUE THREAD

**Issue by sa9sha9** [reviewdog/action-eslint#62]
I guess it should work only with lint **errors**.  Is it possible to fail only on error?

My `reviewdog.yml` is below.
```yml
name: reviewdog
on: [pull_request]
jobs:
  eslint:
    name: runner / eslint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: eslint
        uses: reviewdog/action-eslint@v1
        with:
          reporter: github-pr-review
          workdir: "app/"
          eslint_flags: "--cache --ext .js,.jsx,.ts,.tsx ."
          level: error
          fail_on_error: true
```

**Comment by narrowizard** [comment]
I think it's eslint stuff. add `--quiet` cli option for eslint, and it works well.  
https://eslint.org/docs/user-guide/command-line-interface @sa9sha9

**Comment by hylle** [comment]
@narrowizard With `--quiet` warnings aren't reported at all.
I think what @sa9sha9 is looking for is way to have `error` level findings cause a fail, while still having a `warnings` reported, but without having it cause a fail.

**Comment by carmanchris31** [comment]
> @narrowizard With `--quiet` warnings aren't reported at all. I think what @sa9sha9 is looking for is way to have `error` level findings cause a fail, while still having a `warnings` reported, but without having it cause a fail.

This is the main issue I have with the action also. Eslint only fails when errors are reported and it's reasonable to expect (or at least allow) the same behavior from this action.

**Comment by carmanchris31** [comment]
Looks like this may be a limitation of Github and they have adding support for it in their backlog:
https://github.com/github/feedback/discussions/9875#discussioncomment-2113787

**Comment by carmanchris31** [comment]
Actually It looks like this is already supported:
> The final conclusion of the check. Can be one of action_required, cancelled, failure, neutral, success, skipped, stale, or timed_out
> —[source](https://docs.github.com/en/rest/checks/runs#:~:text=be%20one%20of-,action_required%2C%20cancelled%2C%20failure%2C%20neutral%2C%20success%2C%20skipped%2C%20stale,-%2C%20or%20timed_out.%20When)

Here's an example lint action which is using it:
https://github.com/wearerequired/lint-action/blob/2fe6593ac19ccad08133cf11685d5051fa94bbba/src/github/api.js#L44-L53

**Comment by zachnicoll** [comment]
Would LOVE this to be merged so it behaves as ESLint would behave locally!

**Comment by javierjulio** [comment]
Perhaps worth considering changing the default for a new major version?

**Comment by houserx-jmcc** [comment]
> Perhaps worth considering changing the default for a new major version?

Yeah, this feels like the right path as a default
